platform/linux-generic/drv_driver.c

```diff
@@ -87,6 +95,21 @@ typedef struct _odpdrv_devio_lst_t {
 } _odpdrv_devio_lst_t;
 static struct _odpdrv_devio_lst_t devio_lst;
 
+/* a driver (list element) */
+struct _odpdrv_driver_s {
+	odpdrv_driver_param_t param;
+	_odp_ishm_pool_t *pool;
+	odp_ticketlock_t probelock; /* to avoid concurrent probe on same drv*/
+	struct _odpdrv_driver_s *next;
+};
+
+/* the driver list: */
+typedef struct _odpdrv_driver_lst_t {
+	odp_rwlock_recursive_t lock;
+	_odpdrv_driver_t *head;
+} _odpdrv_driver_lst_t;
+static struct _odpdrv_driver_lst_t driver_lst;
```

Comment:
change only if it doesn't comply with our coding style, I am not sure what applies here